src/vs/workbench/browser/parts/editor/editorGroupsControl.ts

```diff
@@ -2043,7 +2193,16 @@ export class EditorGroupsControl extends Themable implements IEditorGroupsContro
 				}
 			}
 
-			this.visibleEditors[position].layout(new Dimension(editorWidth, editorHeight));
+			editor.getContainer().style({ 'margin-left': editorPosition + 'px', 'width': editorWidth + 'px' });
+			editor.layout(new Dimension(editorWidth, editorHeight));
+		}
+	}
+
+	private resetCenteredEditor(layout: boolean = true) {
+		this.centeredEditorLeftMarginRatio = 0.5;
+		this.centeredEditorPreferedSize = Math.floor(this.dimension.width * 0.5);
```

Comment:
Let's use the [golden ratio](https://en.wikipedia.org/wiki/Golden_ratio) as the default value.
So `this.dimension.width * 0.61`

Reply:
done. nice idea.